npm/qsharp/src/compiler/compiler.ts

```diff
@@ -25,6 +25,13 @@ import {
 // so use those as the set used by the shared compiler
 type Wasm = typeof import("../../lib/web/qsc_wasm.js");
 
+// This type mirrors DocFile in WASM, which is returned from generate_docs()
+export type DocFile = {
+  filename: string;
+  metadata: string;
+  contents: string;
+};
+
```

Comment:
This can easily be generated from the WASM crate to get strong typing and to avoid duplicating types across files like this.

You can apply this commit 
https://github.com/microsoft/qsharp/commit/31a3b31d3ebe4ee785dec90073799cb5473a5406

The generated signature of `this.wasm.generate_docs` will now be `IDocFile[]` instead of `any`. That type can be exported and used in the rest of the code.

Reply:
This is interesting! Thanks for the tip, serializable_type! macro is nice!

Reply:
I took your commit. After our discussion and perf testing this looks good!